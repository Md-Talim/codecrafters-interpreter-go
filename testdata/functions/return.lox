// Test Case 1
fun fib(n) {
  if (n < 2) return n;
  return fib(n - 2) + fib(n - 1);
}

var start = clock();
print fib(32) == 2178309;
print (clock() - start) < 5; // 5 seconds

// Test Case 2
fun f() {
  if (false) return "no"; else return "ok";
}

print f();

// Test Case 3
fun f() {
  while (!true) return "ok";
}

print f();

// Test Case 4
fun f() {
  return;
  print "bad";
}

print f();