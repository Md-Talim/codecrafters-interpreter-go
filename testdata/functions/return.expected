true
true
ok
nil
nil